crates/store/src/db/sql/mod.rs

```diff
@@ -1002,6 +936,54 @@ pub fn select_note_inclusion_proofs(
     Ok(result)
 }
 
+/// Returns a paginated batch of network notes that have not yet been consumed.
+///
+/// # Returns
+///
+/// A set of unconsumed network notes with maximum length of `limit` and a pagination token to get
+/// the next set.
+#[cfg_attr(not(test), expect(dead_code, reason = "gRPC method is not yet implemented"))]
+pub fn unconsumed_network_notes(
+    transaction: &Transaction,
+    mut token: PaginationToken,
+    limit: NonZeroUsize,
+) -> Result<(Vec<NoteRecord>, PaginationToken)> {
+    assert_eq!(
+        NoteExecutionMode::Network as u8,
+        0,
+        "Hardcoded execution value must match query"
+    );
+
+    // Select the rowid column so that we can return a pagination token.
+    //
+    // rowid column _must_ come after the note fields so that we don't mess up the
+    // `NoteRecord::from_row` call.
+    let mut stmt = transaction.prepare_cached(&format!(
+        "
+        SELECT {}, rowid FROM notes
+        WHERE
+            execution_mode = 0 AND consumed = FALSE AND rowid >= ?
+        LIMIT ?
```

Comment:
Question: are the rows automatically sorted by `rowid`? If not, we probably need the `ORDERY BY` clause.

Reply:
Ah. Right.

Thanks, its "undefined". I've added ordering.